build/packer: Configure filebeat installation onto TeamCity agents

spiffyy99 opened the pull request:
Configures installation of [Filebeat](https://www.elastic.co/beats/filebeat) onto dynamically provisioned TeamCity agents for ease of log shipping.

The logs collected by Filebeat are sent to the internal IP of a GCE instance hosting the ELK stack for log parsing and analysis.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50785)
<!-- Reviewable:end -->

r6min commented on build/packer/filebeat/filebeat-agent.yml:
```diff
@@ -0,0 +1,28 @@
+filebeat.inputs:
```
will sync file from [roachsiem](https://github.com/cockroachlabs/roachsiem/) repo.

aaron-crl replied:
Might be worth getting @jlinder  to look at this pattern too since he owns this process at the moment.

r6min replied:
My bad on the terminology side, by 'syncing' I meant we will have to manually update this file whenever there is a change to the file in the roachsiem repo, since it's private and isn't easily accessible on the image.

jlinder replied:
Ah. Ok. How often do you think that will change? (hoping it's not to often because the process for updating this is a pain in the butt).

r6min replied:
Shouldn't be too often, at most whenever there's a change to the way artifacts are released or the logs associated with them. 

This is all of the logic we wanted to monitor build logs for now.

jlinder replied:
Can you add a comment in the file for what needs to be updated when it changes on the roachsiem side? and if there is code on the roachsiem side, point to this file in comments once this file is merged?

r6min replied:
Added comments.

r6min commented on build/packer/filebeat/filebeat-agent.yml:
```diff
@@ -24,5 +24,4 @@ processors:
   timeout: 15
-  # Static Interal IP of ELK instance
-  hosts: ["10.124.0.2:5044"]
+  hosts: ["triptolemus.cockroachlabs.com:5044"]
```
Can use [DNS peering](https://cloud.google.com/dns/docs/overview#dns-peering) to resolve internal hosts across GCP projects.

aaron-crl commented on build/packer/setup_filebeat_on_teamcity_agent.sh:
```diff
@@ -0,0 +1,22 @@
+# Install filebeat service and run at startup
+echo "Installing Filebeat..."
+# Should be latest version of Filebeat (not hard requirement)
+curl -L -o filebeat-amd64.deb https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-7.7.1-amd64.deb
```
Why are we doing this with curl and not APT/YUM?

https://www.elastic.co/guide/en/beats/filebeat/current/setup-repositories.html

r6min replied:
Will test and fix.

r6min replied:
This should also allow us to install the latest.

aaron-crl reviewed (approved):
lgtm

jlinder reviewed (commented):
I left a couple questions and a few change requests. Please also do the following:

- squash all your commits into one commit
- format the commit message [according to the guidelines](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages). An example: 2d91fa0b2718eacac30b0f6ef7a8076a1279621b

r6min commented on build/packer/setup_filebeat_on_teamcity_agent.sh:
```diff
@@ -0,0 +1,29 @@
+apt-get update && apt-get install filebeat
+systemctl enable filebeat
+
+# Copy over config file 
```
Right, so enabling the service at startup is distinct from starting it, as mentioned [here](https://askubuntu.com/questions/733469/what-is-the-difference-between-systemctl-start-and-systemctl-enable). 

This is confusing however, so I'll fix the ordering.

jlinder replied:
Ah! I thought `enable` would also `start`. And `restart` will (presumably) start it if it's not already running. Makes sense. And agreed putting the config file in the right place first still does make more sense.

spiffyy99 committed:
Configure Filebeat installation onto TeamCity agents

build/packer: Dynamically install Filebeat onto TeamCity agents to
easily ship TeamCity build logs for monitoring

Before: The Packer image only had the TeamCity agent service installed
which ran builds from the leader.

Problem: We need a way to install Filebeat onto the agents to easily
ship build logs. Since the agents are dynamically provisioned, Filebeat
needs to be dynamically installed as part of the image setup.

Now: The Packer configuration has been updated to install Filebeat via
a script as well as move the Filebeat config on disk. The logs
collected by Filebeat are sent to a GCP instance running ELK, which is
used to monitor and parse the logs.

This also includes a change to remove non-checked-in files from the
cockroach checkout because the execgen generated files were added into
source control and this caused a caching problem between cockroach
versions made while building the image.

Release Note: None

jlinder reviewed (approved):
LGTM

r6min commented:
bors r+
